packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -1398,6 +1420,10 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
         this._connections = new ec2.Connections({ securityGroups: [this.securityGroup] });
         this.securityGroups = [this.securityGroup];
 
+        if (props.keyPair) {
+          throw new Error('Can only use \'keyPair\' when feature flag \'AUTOSCALING_GENERATE_LAUNCH_TEMPLATE\' is set');
```

Comment:
Do we need this check?

Reply:
I added this validation because if `AUTOSCALING_GENERATE_LAUNCH_TEMPLATE` is disabled, a `CfnLaunchConfiguration` will be created instead `LaunchTemplate` .. and doesn't have support for the keyPair

Reply:
Done! https://github.com/aws/aws-cdk/pull/29679/commits/f7a3896de7612cd80acf23b3b38358ea2a259fb4